File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**6)
count = 1

def dfs(graph, r, visited, order):
    global count
    visited[r] = True         
    order[r] = count             
    count += 1                   

    for i in graph[r]:           
        if not visited[i]:       
            dfs(graph, i, visited, order) 

n, m, r = map(int, input().split())
graph = [[] for _ in range(n + 1)]

visited = [False] * (n + 1)
order = [0] * (n + 1)

for _ in range(m):
    u, v = map(int, input().split())
    graph[u].append(v)
    graph[v].append(u)

for i in range(1, n + 1):
    graph[i].sort()
 
dfs(graph, r, visited, order)

for i in range(1, n + 1):
    print(order[i])

File: 백준/Silver/24479. 알고리즘 수업 － 깊이 우선 탐색 1/알고리즘 수업 － 깊이 우선 탐색 1.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf8").trim().split("\n");

const [N, M, R] = input[0].split(" ").map(Number);
const graph = Array.from({ length: N + 1 }, () => []);
const visited = Array(N + 1).fill(0);

for (let i = 1; i <= M; i++) {
  const [u, v] = input[i].split(" ").map(Number);
  graph[u].push(v);
  graph[v].push(u);
}

for (let i = 1; i <= N; i++) {
  graph[i].sort((a, b) => a - b);
}

let order = 1;

function dfs(node) {
  visited[node] = order++;
  for (const next of graph[node]) {
    if (!visited[next]) dfs(next);
  }
}

dfs(R);

console.log(visited.slice(1).join("\n"));